### PR TITLE
Store dashboard widget IDs

### LINKED ISSUE #301: Store Widget ID In State
Dashboard widget IDs aren't stored in Terraform state, meaning if the widget ID is needed later in the Terraform configuration then there's no straightforward way to obtain it. Would it be possible to add an `id` attribute to the dashboard widget state to assist with retrieving the IDs?

### Terraform Version

```
Terraform v0.12.3
+ provider.datadog v2.2.0
```

### Affected Resource(s)

- datadog_dashboard

### Expected behaviour

The below configuration results in `Success! The configuration is valid.`

```
resource "datadog_dashboard" "ordered_dashboard" {
  title         = "Ordered Layout Dashboard"
  description   = "Created using the Datadog provider in Terraform"
  layout_type   = "ordered"
  is_read_only  = true

  widget {
    alert_graph_definition {
      alert_id = "895605"
      viz_type = "timeseries"
      title = "Widget Title"
      time = {
        live_span = "1h"
      }
    }
  }
}

locals {
  widget_id = datadog_dashboard.ordered_dashboard.widget[0].id
}
```

### Actual Behaviour

```
Error: Unsupported attribute

  on main.tf line 20, in locals:
  20:   widget_id = datadog_dashboard.ordered_dashboard.widget[0].id

This object has no argument, nested block, or exported attribute named "id".
```

### PR DESCRIPTION
**Pull request by therve**
This adds a new attribute to dashboard widgets to store their IDs for
future references.

Closes #301

### CODE REVIEW

**Review by zippolyte — APPROVED** [review]

File: datadog/resource_datadog_dashboard.go [hunk 1]
```diff
@@ -540,6 +540,11 @@ func getNonGroupWidgetSchema() map[string]*schema.Schema {
 				Schema: getWidgetLayoutSchema(),
 			},
 		},
+		"id": {
+			Type:        schema.TypeInt,
+			Computed:    true,
```
Comment (matt-miller-ddog) [inline datadog/resource_datadog_dashboard.go:545]: 👍

File: datadog/resource_datadog_dashboard.go [hunk 2]
```diff
@@ -540,6 +540,11 @@ func getNonGroupWidgetSchema() map[string]*schema.Schema {
 				Schema: getWidgetLayoutSchema(),
 			},
 		},
+		"id": {
+			Type:        schema.TypeInt,
```
Comment (matt-miller-ddog) [inline datadog/resource_datadog_dashboard.go:544]: We've had some trouble in the past with widget IDs big enough to require Int64 when represented in Go. I think `schema.TypeInt` will work (they don't seem to provide an explicitly `Int64` version) but I'm not familiar enough to gauge this properly.

Just wanted to put this on your radar.

**Review by matt-miller-ddog — APPROVED** [review]
Approved, with a comment about `Int64`.

File: datadog/resource_datadog_dashboard.go (hunk 2 above)
Comment (therve) [reply datadog/resource_datadog_dashboard.go:544]: Thanks for the comment: terraform does indeed the right thing, and TypeInt is int64 capable.

**Pull request merged**